packages/access-api/src/models/delegations.js

```diff
@@ -179,3 +195,160 @@ export function delegationsTableBytesToArrayBuffer(sqlValue) {
     return Uint8Array.from(sqlValue)
   }
 }
+
+/**
+ * @typedef {`delegations_v3`} DelegationsTableWithoutBytesName
+ */
+
+/** @type {DelegationsTableWithoutBytesName} */
+export const delegationsV3Table = `delegations_v3`
+
+/**
+ * @typedef {import('../types/access-api-d1').AccessApiD1TablesV2} AccessApiD1TablesV2
+ * @typedef {import('../types/access-api-d1').AccessApiD1TablesV3} AccessApiD1TablesV3
+ */
+
+export class DbDelegationsStorageWithR2 {
+  // @todo abstract away R2 specifics into DagStore: ~AsyncMap<CID, Ucanto.Delegation>
+  /** @type {R2Bucket} */
+  #dags
+  /** @type {import('../types/database').Database<AccessApiD1TablesV3>} */
+  #db
+  /** @type {keyof AccessApiD1TablesV3} */
+  #delegationsTableName = delegationsV3Table
+
+  /**
+   * @param {import('../types/database').Database<AccessApiD1TablesV3>} db
+   * @param {R2Bucket} dags
+   * @param {keyof AccessApiD1TablesV3} delegationsTableName
+   */
+  // eslint-disable-next-line no-useless-constructor
+  constructor(db, dags, delegationsTableName = delegationsV3Table) {
+    this.#db = db
+    this.#delegationsTableName = delegationsTableName
+    this.#dags = dags
```

Comment:
add type assertion that `this` implements `DelegationsStorage` type

Reply:
done https://github.com/web3-storage/w3protocol/pull/578/commits/51e28356016c0eca719be2c0ba3391999270c99f#diff-d9b54b256c8932e4236c04f8c4011720a34850959bf3643834ab025849f8f4f3R227